saleor/payment/gateways/stripe/plugin.py

```diff
@@ -176,14 +176,23 @@ def process_payment(
         data = payment_information.data
 
         payment_method_id = data.get("payment_method_id") if data else None
-        setup_future_usage = data.get("setup_future_usage") if data else None
+
+        setup_future_usage = None
+        if payment_information.reuse_source:
+            setup_future_usage = data.get("setup_future_usage") if data else None
+
         off_session = data.get("off_session") if data else None
 
-        customer = get_or_create_customer(
-            api_key=api_key,
-            customer_email=payment_information.customer_email,
-            customer_id=payment_information.customer_id,
-        )
+        customer = None
+        # confirm that we creates customer on stripe side only for log-in customers
+        # Stripe doesn't allow to search users by email, so each create customer
+        # call creates new customer on Stripe side.
```

Comment:
🚓  Grammar Police - to serve & correct 🚓  
```suggestion
        # Confirm that we create customers in Stripe only for logged-in customers.
        # Stripe doesn't allow to search users by email, so each "create customer"
        # call creates a new customer on Stripe side.
```